twofactor_privacyidea/templates/challenge.php

```diff
@@ -1,15 +1,41 @@
-<?php if($_['messages']): ?>
-<fieldset class="warning">
-<ul>
-<?php foreach($_['messages'] as $message): ?>
-    <li><?php p($message); ?></li>
-<?php endforeach; ?>
-</ul>
-</fieldset>
+<?php
+if ($_["u2fSignRequest"]) {
+    $u2fSignRequest = $_["u2fSignRequest"];
+    script('twofactor_privacyidea', 'u2f-api');
+    script('twofactor_privacyidea', 'u2f');
+}
+?>
+
+<?php if ($_['messages']): ?>
+    <fieldset class="warning">
+        <ul>
+            <?php foreach ($_['messages'] as $message): ?>
+                <li><?php p($message); ?></li>
+            <?php endforeach; ?>
+        </ul>
+        <!--
+        Debug:
+        <?php print_r($_); ?>
+        -->
+    </fieldset>
 <?php endif; ?>
-<form method="POST">
-	<input type="hidden" name="redirect_url" value="<?php p($_['redirect_url']); ?>">
-	<input type="password" name="challenge" placeholder="OTP" autocomplete="off"
-		   autocorrect="off" required autofocus>
-	<input type="submit" class="button" value="Verify">
-</form>
+
+<form method="POST" id="piLoginForm" name="piLoginForm">
+    <input type="hidden" name="redirect_url" value="<?php p($_['redirect_url']); ?>">
+    <input type="hidden" name="transaction_id" value="<?php p($_['transactionId']); ?>">
```

Comment:
We mix different naming conventions here (`transaction_id` vs. `signatureData`), but I'm not sure if we can improve this that much.